data_model/src/lib.rs

```diff
@@ -84,11 +84,11 @@ impl ValidationError {
     DebugCustom, Display, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Encode, Serialize, IntoSchema,
 )]
 #[repr(transparent)]
-pub struct Name(String);
+pub struct Name(ConstString);
 
 impl Name {
     pub(crate) const fn empty() -> Self {
-        Self(String::new())
+        Self(ConstString::new())
```

Comment:
if `Name` is immutable, and I think it is?, this can be a shared const value

Reply:
I don't think about that! Thank you for suggestion.
Do we need empty `Id` in the first place?

Reply:
Yes, we have to support empty `Name`s at at least for now. There is an issue tracking the discussion on whether we'll allow or disallow empty identifiers

Reply:
So the approach has to be one of two. Either we publish this as a separate crate that could be shared with other projects, in which case we need to allow empty values, or we tailor it directly to what our `Name` struct needs (in which case we could equally just merge the two and get rid of another layer of indirection. 

I don't mind either way, but I think that a `ConstString` could be useful in Sora.